src/main/java/com/conveyal/gtfs/loader/conditions/AgencyHasMultipleRowsCheck.java

```diff
@@ -0,0 +1,66 @@
+package com.conveyal.gtfs.loader.conditions;
+
+import com.conveyal.gtfs.error.NewGTFSError;
+import com.conveyal.gtfs.loader.Field;
+import com.conveyal.gtfs.loader.LineContext;
+import com.google.common.collect.HashMultimap;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import static com.conveyal.gtfs.error.NewGTFSErrorType.AGENCY_ID_REQUIRED_FOR_MULTI_AGENCY_FEEDS;
+import static com.conveyal.gtfs.loader.JdbcGtfsLoader.POSTGRES_NULL_TEXT;
+
+/**
+ * Conditional requirement to check that an agency_id has been provided if more than one row exists in agency.txt.
+ */
+public class AgencyHasMultipleRowsCheck extends ConditionalRequirement {
+
+    private final int FIRST_ROW = 2;
+    private final int SECOND_ROW = 3;
+
+    public AgencyHasMultipleRowsCheck(
+        String dependentFieldName,
+        ConditionalCheckType dependentFieldCheck
+    ) {
+        this.dependentFieldName = dependentFieldName;
+        this.dependentFieldCheck = dependentFieldCheck;
+    }
```

Comment:
This conditional requirement is very specific and this constructor is only used in one place so it probably doesn't need to be generalized to have arguments. The `dependentFieldName` and `dependentFieldCheck` could be set as hardcoded values in the constructor body.

Reply:
I ended up implementing this in a proposed refactor: #320